dvc/api.py

```diff
@@ -92,6 +94,175 @@ def read(path, repo=None, rev=None, remote=None, mode="r", encoding=None):
         return fd.read()
 
 
+def get_params(
+    *targets: str,
+    repo: Optional[str] = None,
+    revs: Optional[Union[str, Iterable[str]]] = None,
```

Comment:
I'm not sure if this is worth changing the structure of the returned dict. Seems like it might be better to support a single rev and have other ways to iterate over revs. WDYT?

Reply:
Also, revs passed without repo kwarg returns unexpected results:

```python
>>> dvc.api.get_params(revs="6-prepare-stage")
{'workspace': {'prepare': {'split': 0.2, 'seed': 20170428}, 'featurize': {'max_features': 200, 'ngrams': 2}, 'train': {'seed': 20170428, 'n_est': 50, 'min_split': 0.01}, 'extra': {'foo': 1}}, '6-prepare-stage': {'prepare': {'split': 0.2, 'seed': 20170428}, 'featurize': {'max_features': 100, 'ngrams': 1}, 'train': {'seed': 20170428, 'n_est': 50, 'min_split': 0.01}}}
```

Reply:
> Seems like it might be better to support a single rev and have other ways to iterate over revs. WDYT?

Sounds good

Reply:
> Also, revs passed without repo kwarg returns unexpected results:

So, you would expect `workspace` to not be present?

Reply:
@dberenbaum I cannot reproduce the output of `dvc.api.get_params(revs="6-prepare-stage")` from the current status of the PR. Could you check?

Reply:
Are you planning to remove revs completely or not?

Reply:
Sorry, I realized I already asked the same question above 😬 , and I don't want to block the PR. 

One last concern: 

> ```python
> >>> dvc.api.get_params(revs="6-prepare-stage")
> {'6-prepare-stage': {'prepare': {'split': 0.2, 'seed': 20170428}, 'featurize': {'max_features': 100, 'ngrams': 1}, 'train': {'seed': 20170428, 'n_est': 50, 'min_split': 0.01}}}
> ```

I would like this better if passing a str kept the return dict the same, like:

```python
>>> dvc.api.get_params(revs="6-prepare-stage")
{'prepare': {'split': 0.2, 'seed': 20170428}, 'featurize': {'max_features': 100, 'ngrams': 1}, 'train': {'seed': 20170428, 'n_est': 50, 'min_split': 0.01}}
>>> dvc.api.get_params(revs=["6-prepare-stage"])
{'6-prepare-stage': {'prepare': {'split': 0.2, 'seed': 20170428}, 'featurize': {'max_features': 100, 'ngrams': 1}, 'train': {'seed': 20170428, 'n_est': 50, 'min_split': 0.01}}}
```

Reply:
> can we instead require revs to be an iterable instead of allowing a str? 
🤔 What about `stages`? Do we follow same behavior?

Do you think it would be confusing if I implement as you suggested above:

```
>>> dvc.api.get_params(revs="6-prepare-stage")
{'prepare': {'split': 0.2, 'seed': 20170428}, 'featurize': {'max_features': 100, 'ngrams': 1}, 'train': {'seed': 20170428, 'n_est': 50, 'min_split': 0.01}}
>>> dvc.api.get_params(revs=["6-prepare-stage"])
{'6-prepare-stage': {'prepare': {'split': 0.2, 'seed': 20170428}, 'featurize': {'max_features': 100, 'ngrams': 1}, 'train': {'seed': 20170428, 'n_est': 50, 'min_split': 0.01}}}
```

Reply:
> 🤔 What about `stages`? Do we follow same behavior?

No, I don't think so. That's one reason why I prefer to require `revs` to be an iterable. Then, there is a difference between `stages` and `revs` args that reflects the differences in their behavior.

Changing `stages` will make it harder to use for simple use cases (since passing a single stage is the primary use case). I'm not sure we need to accept an iterable for `stages`, but it's a nice convenience. 

For `revs`, there's no obvious way to merge multiple revs. I'm also not sure there's a simple use case where I want to retrieve params from a single rev outside the current workspace.

> Do you think it would be confusing if I implement as you suggested above:

I don't mind it if you prefer it, but it seems more complicated and makes less clear why we change the return structure for `revs` but not `stages`.

Reply:
Updated to support single `rev`